#  cis--trans-Regulation
Cis and trans regulatory evolution of cotton flowering time control
## Experimental Design
RNA-seq libraries were generated for LD and SD leaf from wild and domesticated Gossypium hirsutum accessions and their reciprocal F1 hybrids each with three replicates under long-day and short-day conditions:

**·** 2 condition: LD (7am & 9pm)& SD (7am &5 pm)
 
**·** 4 cotton accessions: Maxxa, TX2094, reciprocal F1 hybrids MxT and TxM

**·** 3-6 biological replicates
## RNA-seq mapping and high-quality SNP identification between parents
Filtered RNA-seq reads were mapped against G. hirsutum(TM-1) to estimate gene expressions in different accessions. In order to distinguish Maxxa(Ma) and TX2094(Tx), we used GATK(4.0.0.0) to detect allelic SNPs.

### 1.SNP calling
In order to obtain high-confidence allelic SNPs between parents, the fiber transcriptome sequencing data (Bao et al. 2019) was integrated into this analysis.





